pkg/awsclient/ec2_helper.go

```diff
@@ -49,13 +49,15 @@ Outer:
 			*ingressRule.IpProtocol != "tcp" {
 			continue
 		}
-		for _, cidrBlock := range cidrBlocks {
-			// Note: For now, we assume that ingressRule.IpRange is length 1 as that
-			// appears to be the usage inside AWS.
-			if *ingressRule.IpRanges[0].CidrIp == cidrBlock {
-				seenExpectedRules[cidrBlock] = true
-				// No need to continue on this ingressRule, because we seen it
-				continue Outer
+		// Since each ingressRule can have 0 or more IP Ranges defined, iterate through
+		// them for matches
+		for _, ipRange := range ingressRule.IpRanges {
+			for _, cidrBlock := range cidrBlocks {
+				if *ipRange.CidrIp == cidrBlock {
+					seenExpectedRules[cidrBlock] = true
+					// No need to continue on this ingressRule, because we seen it
+					continue Outer
```

Comment:
what is Outer here?

Reply:
`Outer` refers to the label on line 44, which is a bit outside the scope of what shows up in the diff.

Semantically it means: "If we have seen a CIDR block [that we want present] in the list that AWS gave us, then we can stop looking for it, and begin the next CIDR block [we want present]"

Reply:
ah makes sense. I didn't see the label